mpmath/tests/test_format.py

```diff
@@ -644,6 +644,34 @@ def test_mpf_fmt():
         assert f"{mp.mpf('-123.456'):.5Zg}" == "-123.45"
         assert f"{mp.mpf('-123.456'):.5Ng}" == "-123.46"
 
+        num = mp.mpf('0.1')
+        assert f"{-num:=.2Df}" == "-0.10"
```

Comment:
Are you sure?  Looks incorrect for me, c.f.:
```pycon
>>> import gmpy2
>>> num = gmpy2.mpfr('0.1')
>>> f"{-num:.2Df}"
'-0.11'
```

Reply:
I am sure about rounding 0 down. What you show me is because of a different issue:
```python
>>> from gmpy2 import mpfr
>>> print('{:.30f}'.format(mpfr(0.1)))
0.10000000000000005551115123126
>>> print('{:.2Uf}'.format(mpfr(0.1)))
0.11
```
Here 0.1 is rounded up because the incomplete floating point representation is slightly higher than 0.1.
On the other hand, if you format an exact float,
```python
>>> print('{:.30f}'.format(mpfr(0.25)))
0.250000000000000000000000000000
>>> print('{:.3Uf}'.format(mpfr(0.25)))
0.250
```
the 0 gets rounded down.

If I do the same with mpmath, leaving the code as it was before, I get
```python
>>> print('{:.2Uf}'.format(mp.mpf('0.1')))
0.11
>>> print('{:.3Uf}'.format(mp.mpf('0.25')))
0.251
```
whereas with the new one,
```python
>>> from mpmath import mp
>>> print('{:.2Uf}'.format(mp.mpf('0.1')))
0.10
>>> print('{:.3Uf}'.format(mp.mpf('0.25')))
0.250
```

The way I coded `round_digits()`, it only looks at the digit that is next to the one we round to. I should do something similar to what I did for tying `round_nearest` to even. The incorrect rounding up of the digit 0  was hiding this other problem.

Reply:
> Here 0.1 is rounded up because the incomplete floating point representation is slightly higher than 0.1.

What you mean by "incomplete"?  Your rounding algorithm should work with binary floating-point numbers, not only with whose decimal numbers, that are binary fractions, like 0.25.

> If I do the same with mpmath, leaving the code as it was before, I get

Unfortunately, this fix breaks other tests, shown above.  We can't include wrong tests in CI.

> I should do something similar to what I did for tying round_nearest to even.

That sounds much better.

Yes, you should examine all digits after dps, not just assume they are ``0``.